delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/ElasticSearchViews.scala

```diff
@@ -1,21 +1,254 @@
+import ch.epfl.bluebrain.nexus.delta.sdk.jsonld.JsonLdSourceParser
+import ch.epfl.bluebrain.nexus.delta.sdk.jsonld.JsonLdSourceParser.expandIri
+import ch.epfl.bluebrain.nexus.delta.sdk.model.IdSegment.StringSegment
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Identity.Subject
 import ch.epfl.bluebrain.nexus.delta.sdk.model.permissions.Permission
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.{Project, ProjectRef}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.Pagination.FromPagination
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.ResultEntry.UnscoredResultEntry
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.SearchResults.UnscoredSearchResults
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{Envelope, IdSegment, Label}
 import ch.epfl.bluebrain.nexus.delta.sdk.utils.{IOUtils, UUIDF}
+import ch.epfl.bluebrain.nexus.sourcing.processor.EventSourceProcessor.persistenceId
+import ch.epfl.bluebrain.nexus.sourcing.{Aggregate, EventLog}
 import io.circe.Json
-import monix.bio.{IO, UIO}
+import io.circe.syntax._
+import monix.bio.{IO, Task, UIO}
 
-class ElasticSearchViews {}
+final class ElasticSearchViews private (
+    aggregate: ElasticSearchViewAggregate,
+    eventLog: EventLog[Envelope[ElasticSearchViewEvent]],
+    cache: ElasticSearchViewCache,
+    projects: Projects
+)(implicit rcr: RemoteContextResolution, uuidF: UUIDF) {
+
+  def create(
+      project: ProjectRef,
+      value: ElasticSearchViewValue
+  )(implicit subject: Subject): IO[ElasticSearchViewRejection, ElasticSearchViewResource] =
+    uuidF().flatMap(uuid => create(StringSegment(uuid.toString), project, value))
+
+  def create(
+      id: IdSegment,
+      project: ProjectRef,
+      value: ElasticSearchViewValue
+  )(implicit subject: Subject): IO[ElasticSearchViewRejection, ElasticSearchViewResource] =
+    create(id, project, value, value.asJson)
+
+  def create(
+      project: ProjectRef,
+      source: Json
+  )(implicit subject: Subject): IO[ElasticSearchViewRejection, ElasticSearchViewResource] = {
+    for {
+      p          <- projects.fetchActiveProject[ElasticSearchViewRejection](project)
+      result     <- JsonLdSourceParser.decode[ElasticSearchViewValue, ElasticSearchViewRejection](p, source)
+      (id, value) = result
+      res        <- eval(CreateElasticSearchView(id, project, value, source, subject), p)
+    } yield res
+  }.named("createElasticSearchView", moduleType)
+
+  def create(
+      id: IdSegment,
+      project: ProjectRef,
+      source: Json
+  )(implicit subject: Subject): IO[ElasticSearchViewRejection, ElasticSearchViewResource] = {
+    for {
+      p     <- projects.fetchActiveProject(project)
+      iri   <- expandIri(id, p)
+      value <- JsonLdSourceParser.decode[ElasticSearchViewValue, ElasticSearchViewRejection](p, iri, source)
+      res   <- eval(CreateElasticSearchView(iri, project, value, source, subject), p)
+    } yield res
+  }.named("createElasticSearchView", moduleType)
```

Comment:
Should the 3 create operations have different names ?

Reply:
If that's the case, we have to change it everywhere. So far, there are no different names

Reply:
Maybe just have two different names when we decode the json and when we are not